File: Design-Patterns/Strategy/Cpp/warrior_attribute_strategy.cpp
#include "attribute_strategy.h"

int warrior_attribute_strategy::get_health()
{
	return properties_->level * properties_->strength * 5;
}

int warrior_attribute_strategy::get_mana()
{
	return properties_->level * properties_->wisdom;
}

int warrior_attribute_strategy::get_attack()
{
	return properties_->level + properties_->strength * 10;
}

int warrior_attribute_strategy::get_magical_attack()
{
	return properties_->level + properties_->wisdom;
}

int warrior_attribute_strategy::get_defense()
{
	return properties_->level + properties_->strength * 5;
}

int warrior_attribute_strategy::get_magical_defense()
{
	return properties_->level + properties_->wisdom;
}

File: Design-Patterns/Strategy/CSharp/AttributeStrategies/WarriorAttributeStrategy.cs
namespace Strategy.AttributeStrategies
{
    public class WarriorAttributeStrategy : AttributeStrategy
    {
        public override int Health => Properites.Level * Properites.Strength * 5;
        public override int Mana => Properites.Level * Properites.Wisdom;
        public override int Attack => Properites.Level + Properites.Strength * 10;
        public override int MagicalAttack => Properites.Level + Properites.Wisdom;
        public override int Defense => Properites.Level + Properites.Strength * 5;
        public override int MagicalDefense => Properites.Level + Properites.Wisdom;
    }
}
